### ISSUE TITLE
A3 Q1d: np.random.seed what does it do?

### ISSUE THREAD

**Issue by rennmeli** [BME1478H/Winter2020class#40]
I'm trying to add noise to the population_size at each point in time. In doing so I used the np.random.normal expression to generate a random value from a normal distribution of mean = 0 and stdev = 250. However, when I try and check that the noise produced has a mean and std deviation of 0 and 250 respectively, I am getting different values (-56.3584 and 231.0712). Please advise. 

Also,

(1) What is the difference between using np.random.seed(42) and np.random.seed(0)?
(2) Is it correct to be specifying a size of 50 for the np.random.normal as the population_size array has 50 as well? 
 
![Screen Shot 2020-01-22 at 4 11 43 PM](https://user-images.githubusercontent.com/59658814/72935163-ce7a4480-3d32-11ea-85e6-c993b1a1fdfa.png)

**Comment by RayNele** [comment]
the noise you are adding is working as intended. `np.random.normal` selects a random number from a normal distribution with specified mean and stdev. In this case mean = 0 and stdev = 250, values of (-56.3584 and 231.0712) are not unusual at all; they are within one standard deviation. In fact, even values up to 750 I would not consider unusual. Since this isn't a statistics course, I can't get much further into normal distributions but you're welcome to look it up on your own time (~5 minute read on wikipedia)

1) There is no difference that would effect your assignment. When `np.random` generates random values, it would be nice for us to be able to keep those values the same (but still random) among multiple separate trials right? That's what a seed does. having a seed of 0 and 42 will generate equally random numbers, but if you ever want to generate those same random numbers again, you'll need to have that seed. For example, if I went onto my python, and generated 10 random numbers with `np.random.randint(1,10,10)`, I could run that over and over and over and get different numbers every time. But if I specify a seed: `np.random.seed(42)` before `np.random.randint(1,10,10)` I'll get the same output every time: `array([7, 4, 8, 5, 7, 3, 7, 8, 5, 4])` in fact if you do this exact same thing in your python on your machine, you'll get the same numbers as me as well. This is very helpful for testing behaviour for certain pieces of code.  (quick and dirty explanation, but if that wasn't sufficient I'm happy to delve further into it)

2)Yes it is correct to be specifying that, since we want to be adding actual noise, and not just a single random number to every element of the array. What I mean by this is, if we don't specify a size of 50, your `np.random.normal` will pick one value (say 100), and it will add that to every single datapoint. That's not much for noise..... 

If we specify a size of 50, it will pick 50 different numbers along that distribution, and add those to their respective elements. Element 0 of the data array gets added to element 0 of the noise array, element 1 of the data array gets added to element 1 of the noise array, etc. That way we actually have random noise, and not some static value that was added to every element. 

However, if we "hardcode" the 50 value in there, any data of size > 50 wouldn't work, so I would suggest making the size of the noise array variable against the size of the data array with the use of the `len` command.
